[IMP] default_warehouse_from_sale_team: hook to fill allowed sales teams T#79454

luisg123v opened the pull request:
Implement a hook to fill allowed sales teams in users that are already
a member of any team.

Since this module implements a feature to restrict which sales teams a
user may be a member of, users that already belong to any team are
configured to be allowed for those teams, to avoid
inconsistencies between allowed and already-configured memberships. In
other words, if a user already belongs to a team, it most likely means
they should be allowed to belong to it, so allowance is granted.

xmglord commented on default_warehouse_from_sale_team/hooks.py:
```diff
@@ -0,0 +1,25 @@
+
+
+def fill_user_allowed_salesteams(env):
+    """Fill allowed salesteams in users that are already members of any team
```
missing dot at the end of line

luisg123v replied:
TL;DR
My understanding was 1st line of a multi-line docstring shouldn't include a trailing dot, similar as commit titles. However, by reading [the docstring conventions for multi-line comments](https://peps.python.org/pep-0257/#multi-line-docstrings), I see, while not explicitly mentioned, the provided example includes it. For this, I've changed it.

Short answer: applied, thanks!

xmglord replied:
Thanks to you for the docstring conventions link.

xmglord commented on default_warehouse_from_sale_team/hooks.py:
```diff
@@ -0,0 +1,25 @@
+def fill_user_allowed_salesteams(env):
+    """Fill allowed salesteams in users that are already members of any team
+
+    Since this module implements a feature to restrict what sales teams a user may be member of, users that already
```
what sales teams -> **which** sales teams
a user may be member of -> a user may be **a** member of
inconcistencies -> incon**s**istencies

luisg123v replied:
Sorry, I didn't see this comment, applied. I modified the whole message for clarity.

BTW, I don't think replacing "what" by "which" is necessary on this case, but applied anyway.

xmglord replied:
Since is not necessary lets say it was a nitpick

xmglord commented:
commit description:
f.ill -> fill
missing final dot at the first paragraph
same comments as in docstring

luisg123v commented:
@xmglord Fixes to commit messages applied.

xmglord commented:
> @xmglord Fixes to commit messages applied.

Not applied yet

luisg123v committed:
[IMP] default_warehouse_from_sale_team: hook to fill allowed sales teams

Implement a hook to fill allowed sales teams in users that are already
a member of any team.

Since this module implements a feature to restrict which sales teams a
user may be a member of, users that already belong to any team are
configured to be allowed for those teams, to avoid
inconsistencies between allowed and already-configured memberships. In
other words, if a user already belongs to a team, it most likely means
they should be allowed to belong to it, so allowance is granted.

xmglord reviewed (approved):
LGTM :+1:

TeposteAJ commented:
LGTM :+1:

ajteposte reviewed (approved):
LGTM 👍
